modules/fluid_properties/src/fluidproperties/IdealRealGasMixtureFluidProperties.C

```diff
@@ -14,8 +14,224 @@
+    std::vector<Real> dy_dx_Tv;                                                                    \
+    prop##_from_T_v(T, v, x, y, dy_dT_v, dy_dv_T, dy_dx_Tv);                                       \
+                                                                                                   \
+    dy_dp = dy_dv_T * dv_dp;                                                                       \
+    dy_dT = dy_dT_v + dy_dv_T * dv_dT;                                                             \
+    dy_dx.resize(_n_secondary_vapors);                                                             \
+    for (unsigned int i = 0; i < _n_secondary_vapors; i++)                                         \
+      dy_dx[i] = dy_dx_Tv[i] + dy_dv_T * dv_dx[i];                                                 \
+  }
+
+/**
+ * Defines the value and derivative methods from (T,v) for a mass-specific property y
+ */
+#define define_mass_specific_prop_from_T_v(prop)                                                   \
+  Real IdealRealGasMixtureFluidProperties::prop##_from_T_v(                                        \
+      Real T, Real v, const std::vector<Real> & x) const                                           \
+  {                                                                                                \
+    const Real x_primary = primaryMassFraction(x);                                                 \
+    Real y = x_primary * _fp_primary->prop##_from_T_v(T, v / x_primary);                           \
+                                                                                                   \
+    for (unsigned int i = 0; i < _n_secondary_vapors; i++)                                         \
+      y += x[i] * _fp_secondary[i]->prop##_from_T_v(T, v / x[i]);                                  \
+                                                                                                   \
+    return y;                                                                                      \
+  }                                                                                                \
+                                                                                                   \
+  void IdealRealGasMixtureFluidProperties::prop##_from_T_v(Real T,                                 \
+                                                           Real v,                                 \
+                                                           const std::vector<Real> & x,            \
+                                                           Real & y,                               \
+                                                           Real & dy_dT,                           \
+                                                           Real & dy_dv,                           \
+                                                           std::vector<Real> & dy_dx) const        \
```

Comment:
do we ever care about dy_dxprimary?

Reply:
It depends on the flow formulation, but in any case, adding it would be redundant due to sum-to-unity.